### PR TITLE
cve-bin-tool: unstable-2021-04-15 -> 2.2.1

### PR DESCRIPTION
**Pull request by akshatagarwl**
* move cve-bin-tool from development/python-modules to tools/security

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/cve-bin-tool/default.nix [hunk 1]
```diff
@@ -1,45 +1,22 @@
 { lib
-, buildPythonPackage
+, python3Packages
```
Comment (r-burns) [inline pkgs/tools/security/cve-bin-tool/default.nix:2]:
```suggestion
, buildPythonApplication
```

File: pkgs/tools/security/cve-bin-tool/default.nix [hunk 2]
```diff
@@ -1,45 +1,22 @@
...
 , fetchFromGitHub
-, jsonschema
```
Comment (r-burns) [inline pkgs/tools/security/cve-bin-tool/default.nix:4]: Keep these python3 packages in the args, they're easier to override this way

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -4168,6 +4168,8 @@ with pkgs;
 
   cwebbin = callPackage ../development/tools/misc/cwebbin { };
 
+  cve-bin-tool = callPackage ../tools/security/cve-bin-tool { };
```
Comment (r-burns) [inline pkgs/top-level/all-packages.nix:4171]:
```suggestion
  cve-bin-tool = with python3Packages; callPackage ../tools/security/cve-bin-tool { };
```

File: pkgs/tools/security/cve-bin-tool/default.nix [hunk 4]
```diff
@@ -1,45 +1,22 @@
...
-    sha256 = "STf0tJBpadBqsbC+MghBai8zahDkrXfLoFRJ+84wvvY=";
+    rev = "v${version}";
+    sha256 = "087w7fsc4vd4sjz8ww6q71b108yhz94ydr76d99rhlmcqsq7fihs";
   };
 
   # Wants to open a sqlite database, access the internet, etc
   doCheck = false;
 
-  propagatedBuildInputs = [
+  propagatedBuildInputs = with python3Packages; [
```
Comment (r-burns) [inline pkgs/tools/security/cve-bin-tool/default.nix:19]:
```suggestion
  propagatedBuildInputs = [
```

**Review by r-burns — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (cole-h) [reply pkgs/top-level/all-packages.nix:4171]: No need to use `with`.

```suggestion
  cve-bin-tool = python3Packages.callPackage ../tools/security/cve-bin-tool { };
```

**Comment by humancalico** [comment]
@r-burns @cole-h PTAL

**Review by r-burns — APPROVED** [review]
I think this is correct, it looks like cve-bin-tool is only intended to be used from the CLI

**Comment by DavHau** [comment]
Tested on x86_64-linux. It builds and all resulting executables seem to work.

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 138750` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>cve-bin-tool</li>
  </ul>
</details>

**Comment by happysalada** [comment]
So the darwin eval is failing with 
```
            other_exceptions = [str(e) for e in rejections]
>           raise ParserRejectedMarkup(
                "The markup you provided was rejected by the parser. Trying a different parser or a differ
            )
E           bs4.builder.ParserRejectedMarkup: The markup you provided was rejected by the parser. Trying a
E
E           Original exception(s) from parser:
E            UnicodeDecodeError: 'utf-8' codec can't decode byte 0x91 in position 0: invalid start byte
E            LookupError: unknown encoding: 'b'windows-1252''

bs4/__init__.py:371: ParserRejectedMarkup
_________ LXMLXMLTreeBuilderSmokeTest.test_can_parse_unicode_document __________

self = <bs4.tests.test_lxml.LXMLXMLTreeBuilderSmokeTest testMethod=test_can_parse_unicode_document>

    def test_can_parse_unicode_document(self):
        markup = '<?xml version="1.0" encoding="euc-jp"><root>Sacr\N{LATIN SMALL LETTER E WITH ACUTE} bleu
        soup = self.soup(markup)
>       self.assertEqual('Sacr\xe9 bleu!', soup.root.string)
E       AttributeError: 'NoneType' object has no attribute 'string'

bs4/testing.py:1002: AttributeError
=========================== short test summary info ============================
FAILED bs4/tests/test_lxml.py::LXMLTreeBuilderSmokeTest::test_real_hebrew_document
FAILED bs4/tests/test_lxml.py::LXMLTreeBuilderSmokeTest::test_smart_quotes_converted_on_the_way_in
FAILED bs4/tests/test_lxml.py::LXMLXMLTreeBuilderSmokeTest::test_can_parse_unicode_document
======================== 3 failed, 514 passed in 7.65s =========================
```

do we remove darwin from the platforms or disable the tests ?

**Comment by DavHau** [comment]
Here an example on disabling individual tests for python builds: https://github.com/NixOS/nixpkgs/blob/bdbded8cc1a2b8e324e8c8cfb9617aa6616b4ac6/pkgs/applications/backup/vorta/default.nix#L70-L85

`pytestCheckHook` needs to be in `checkInputs` for this to work.

**Comment by SuperSandro2000** [comment]
> So the darwin eval is failing with

*build

but it is darwin, so it is not a blocker.

**Pull request merged**